Components/Document/InvoiceDocumentHandler.php

```diff
@@ -45,36 +45,47 @@ public function __construct(
      */
     public function handleDocument($orderNumber, Document $document)
     {
-        //Collect all available containers in order to work with some of them later.
-        /** @var array $templateContainers */
-        $templateContainers = $document->_view->getTemplateVars('Containers');
-        /** @var \Smarty_Data $view */
-        $view = $document->_view;
-        /** @var array $orderData */
-        $orderData = $view->getTemplateVars('Order');
-        $orderData = $this->overwritePaymentName($orderData);
-
-        //Get the new footer for the document and replace the original one
-        $rawFooter = $this->getInvoiceContainer($templateContainers, $orderData);
-        $templateContainers['PayPal_Unified_Instructions_Content']['value'] = $rawFooter['value'];
-
-        $view->assign('Order', $orderData);
-        $view->assign('Containers', $templateContainers);
-
-        $instructions = $this->instructionService->getInstructions($orderNumber);
-        if ($instructions) {
-            $document->_template->assign('PayPalUnifiedInvoiceInstruction', $instructions->toArray());
+        // replace the contents only for that document types,
+        // whose really has PayPal boxes assigned to it:
+        if ($this->hasPayPalUnifiedBoxes($document)) {
```

Comment:
You could split it into an early return

Reply:
Yeah, I already thought of this. But as far as I know, early returns are a bad code style pattern, unless it is really necessary. Also I noticed, that you guys like nice formatted code. So, I prefered better readability over early returns.